### ISSUE TITLE
Bug: `@typescript-eslint/prefer-optional-chain` should be smarter with AND `&&`

### ISSUE THREAD

**Issue by B2o5T** [typescript-eslint/typescript-eslint#6435]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=DYUwLgBAZgliwBMAiIoC4IEMB2BPAUPjFBABQAOmATiNmAPwB02mAtiBAGSf4TRyIUJbr37xkqZmw4i+scUKnsIAQgC8aiAHIA+jrC5ytaVq485AiVCUd1m3ToDOAYwAWIVpi0BKCAG8IAF8gA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloHsalfkwCG8WmQAWo5uki9o-aJHBgAviHVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
if (parent?.name &&
  fieldDef &&
  fieldDef.name &&
  fieldDef.name !== '__typename' &&
  fieldDef.name !== '__schema') {}
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/prefer-optional-chain":"error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

I expected `fieldDef && fieldDef.name` to replace by `fieldDef?.name`

### Actual Result

no errors

**Comment by bradzacher** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/6332 (your own issue!) please use the search

**Comment by B2o5T** [comment]
yeah but it is another case with `AND` and not `OR` 😅
